addons/l10n_ar_withholding/models/account_payment.py

```diff
@@ -7,3 +7,18 @@ class AccountPayment(models.Model):
     _inherit = 'account.payment'
 
     l10n_ar_withholding_ids = fields.One2many(related='move_id.l10n_ar_withholding_ids')
+
+    def _synchronize_to_moves(self, changed_fields):
+        ''' If we change a payment with withholdings, delete all withholding lines as the synchronization mechanism is not
+        implemented yet
+        '''
+        if not any(field_name in changed_fields for field_name in self._get_trigger_fields_to_synchronize()):
+            return
+        for pay in self:
+            pay.line_ids.filtered(
```

Comment:
we should filter the lines from the move
`            pay.move_id.line_ids.filtered(
`